src/main/groovy/io/gatling/gradle/GatlingPluginExtension.groovy

```diff
@@ -113,33 +130,109 @@ class GatlingPluginExtension implements JvmConfigurable {
             return packageId
         }
 
+        @Input
+        @Optional
+        UUID getTeamId() {
+            return teamId
+        }
+
         @Input
         @Optional
         URL getUrl() {
             return url
         }
 
+        URL getApiUrl() {
+            return new URL(url.toExternalForm() + PUBLIC_API_PATH)
+        }
+
         @Input
         @Optional
         String getSimulationClass() {
             return simulationClass
         }
 
-        EnterpriseClient getEnterpriseClient(String version) {
-            try {
-                def apiToken = getApiToken()
-                if (!apiToken) {
-                    throw new IllegalArgumentException("The setting gatling.enterprise.apiToken needs to be set, please see https://gatling.io/docs/enterprise/cloud/reference/admin/api_tokens/ to create an API token")
+        @Input
+        @Optional
+        boolean getBatchMode() {
+            return batchMode
+        }
+
+        EnterpriseClient initEnterpriseClient(String version, Logger logger) {
+            def apiToken = getApiToken()
+            if (!apiToken) {
+                throw new IllegalArgumentException("The setting gatling.enterprise.apiToken needs to be set, please see https://gatling.io/docs/enterprise/cloud/reference/admin/api_tokens/ to create an API token")
```

Comment:
Specify system property (following same pattern as other plugins)
Example: 

> A package ID is required to upload a package on Gatling Enterprise; see https://gatling.io/docs/enterprise/cloud/reference/user/package_conf/, create a package and copy its ID.
> You can then set your package ID value by passing it with -Dgatling.enterprise.packageId=\<packageId\>, or add the configuration to your SBT settings, e.g.:
> gatling.enterprise.apiToken MY_PACKAGE_ID

Reply:
question: when does `TaskExecutionException` is needed? (why not here)

Reply:
TaskExecutionException is not needed, but it is cleaner to use it

Reply:
Ok, so not blocking for me but maybe add an inner TaskArgumentException class to throw it each time without too much overhead?